src/Spark.Engine.Test/Service/PatchServiceTests.cs

```diff
@@ -174,6 +186,57 @@ public void CanApplyCollectionAddPatchForNonNamedDataTypes()
             Assert.Equal("testProcessing", resource.Processing[0].Description);
             Assert.Equal(dateTime, resource.Processing[0].Time);
         }
+        
+        [Fact]
+        public void CanApplyCollectionAddPatchForNonNamedDataTypesWithExtension()
```

Comment:
Good idea. I did not, but I will check with unit tests.

Reply:
Checked and added unit tests. Found one potential issue and also fixed it.
Regarding insert. I am not sure current code handles it according to spec. We seems to be getting indexes from path, while specification says that the index should be taken from index part. (https://www.hl7.org/fhir/fhirpatch.html). I did add the handling of this case for the insert, without removing the current logic, but if it is not according to spec it might be better to remove it, IMHO. I also marked it with ToDo in unit tests. (maybe I should have done it in the code.) The examples for the insert could be found at FHIR site: (https://www.hl7.org/fhir/test-cases.zip). I looked at patch\fhir-path-tests.xml.

Reply:
Agree, that should not be correct to how the current specification is to be interpreted. If you can remove it I would appreciate it, better to remove it now before people start relying heavily on it.